src/kernels/jupyter/launcher/jupyterExecution.ts

```diff
@@ -278,8 +269,11 @@ export class JupyterExecutionBase implements IJupyterExecution {
         customCommandLine: string[],
         workingDirectory: Uri,
         cancelToken: CancellationToken
-    ): Promise<IJupyterConnection | undefined> {
-        return this.notebookStarter?.start(
+    ): Promise<IJupyterConnection> {
+        if (!this.notebookStarter) {
```

Comment:
This isn't possible in desktop. hence throw an error.

Reply:
Doesn't need to be localized, its basically a bug in the code that would cause the entire extensoin to go kaboom.
Wish there was a better way to handle such types, these nullables really do complicate the code.

Reply:
Hopefully this will get fixed with the refactoring I'm working on.